istio/istio: `BUILD_WITH_CONTAINER=1 make sidecar.deb` fails on MacOS

**Bug description**
Trying `BUILD_WITH_CONTAINER=1 make sidecar.deb` on MacOS fails:
```
Building with the build container: gcr.io/istio-testing/build-tools:master-2019-12-15T16-17-48.
Using docker credential directory /Users/ericvn/.docker.
Using local Kubernetes configuration /Users/ericvn/.kube
ISTIO_OUT=/work/out/darwin_amd64 ISTIO_BIN=/gobin bin/init.sh
Skipping envoy debug. Set DEBUG_IMAGE to download.
/work/out/linux_amd64/release /work
Downloading Envoy: curl -fLSs https://storage.googleapis.com/istio-build/proxy/envoy-alpha-e4df956fb629490dbc7af43ec9c5edda3245d45f.tar.gz to /work/out/linux_amd64/release/envoy-e4df956fb629490dbc7af43ec9c5edda3245d45f

real	0m2.947s
user	0m0.644s
sys	0m0.672s
Copying /work/out/linux_amd64/release/envoy-e4df956fb629490dbc7af43ec9c5edda3245d45f to /work/out/linux_amd64/release/envoy
/work
Copying /work/out/linux_amd64/release/envoy-e4df956fb629490dbc7af43ec9c5edda3245d45f to /work/out/darwin_amd64/envoy
Copying /work/out/darwin_amd64/envoy to /gobin/envoy
touch /work/out/darwin_amd64/istio_is_init
mkdir -p /work/out/darwin_amd64/logs
make: *** No rule to make target '/work/out/darwin_amd64/pilot-agent', needed by '/work/out/darwin_amd64/release/istio-sidecar.deb'.  Stop.
make: *** [sidecar.deb] Error 2
```



**Expected behavior**
`BUILD_WITH_CONTAINER=1 make sidecar.deb` should work on MacOS. Note that one does need to build using the container as some linux function is needed.

**Steps to reproduce the bug**
Prepare MacOS development as specified.

Run `BUILD_WITH_CONTAINER=1 make sidecar.deb`

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

N/A

**How was Istio installed?**

N/A

**Environment where bug was observed (cloud vendor, OS, etc)**

macOS Catalina

--- a/tools/packaging/deb/istio.mk
+++ b/tools/packaging/deb/istio.mk
@@ -13,9 +13,9 @@ deb/build-in-docker:
 # This target uses a locally installed 'fpm' - use 'docker.sidecar.deb' to use
 # the builder image.
 # TODO: consistent layout, possibly /opt/istio-VER/...
-sidecar.deb: ${ISTIO_OUT}/release/istio-sidecar.deb
+sidecar.deb: ${ISTIO_OUT_LINUX}/release/istio-sidecar.deb
 
-deb: ${ISTIO_OUT}/release/istio-sidecar.deb
+deb: ${ISTIO_OUT_LINUX}/release/istio-sidecar.deb
 
 # Base directory for istio binaries. Likely to change !
 ISTIO_DEB_BIN=/usr/local/bin
@@ -23,14 +23,14 @@ ISTIO_DEB_BIN=/usr/local/bin
 ISTIO_DEB_DEPS:=pilot-discovery istioctl mixs istio_ca
 ISTIO_FILES:=
 $(foreach DEP,$(ISTIO_DEB_DEPS),\
-        $(eval ${ISTIO_OUT}/release/istio.deb: $(ISTIO_OUT)/$(DEP)) \
-        $(eval ISTIO_FILES+=$(ISTIO_OUT)/$(DEP)=$(ISTIO_DEB_BIN)/$(DEP)) )
+        $(eval ${ISTIO_OUT_LINUX}/release/istio.deb: $(ISTIO_OUT_LINUX)/$(DEP)) \
+        $(eval ISTIO_FILES+=$(ISTIO_OUT_LINUX)/$(DEP)=$(ISTIO_DEB_BIN)/$(DEP)) )
 
-SIDECAR_DEB_DEPS:=envoy pilot-agent node_agent istio-iptables istio-clean-iptables
+SIDECAR_DEB_DEPS:=release/envoy pilot-agent node_agent istio-iptables istio-clean-iptables
 SIDECAR_FILES:=
 $(foreach DEP,$(SIDECAR_DEB_DEPS),\
-        $(eval ${ISTIO_OUT}/release/istio-sidecar.deb: $(ISTIO_OUT)/$(DEP)) \
-        $(eval SIDECAR_FILES+=$(ISTIO_OUT)/$(DEP)=$(ISTIO_DEB_BIN)/$(DEP)) )
+        $(eval ${ISTIO_OUT_LINUX}/release/istio-sidecar.deb: $(ISTIO_OUT_LINUX)/$(DEP)) \
+        $(eval SIDECAR_FILES+=$(ISTIO_OUT_LINUX)/$(DEP)=$(ISTIO_DEB_BIN)/$(DEP)) )
 
 ISTIO_DEB_DEST:=${ISTIO_DEB_BIN}/istio-start.sh \
 		${ISTIO_DEB_BIN}/istio-node-agent-start.sh \
@@ -39,7 +39,7 @@ ISTIO_DEB_DEST:=${ISTIO_DEB_BIN}/istio-start.sh \
 		/var/lib/istio/envoy/sidecar.env
 
 $(foreach DEST,$(ISTIO_DEB_DEST),\
-        $(eval ${ISTIO_OUT}/istio-sidecar.deb:   tools/packaging/common/$(notdir $(DEST))) \
+        $(eval ${ISTIO_OUT_LINUX}/istio-sidecar.deb:   tools/packaging/common/$(notdir $(DEST))) \
         $(eval SIDECAR_FILES+=${REPO_ROOT}/tools/packaging/common/$(notdir $(DEST))=$(DEST)))
 
 SIDECAR_FILES+=${REPO_ROOT}/tools/packaging/common/envoy_bootstrap_v2.json=/var/lib/istio/envoy/envoy_bootstrap_tmpl.json
@@ -53,12 +53,12 @@ ISTIO_DEB_NAME ?= istio-sidecar
 # a /etc/systemd/system/multi-user.target.wants/istio.service and auto-start. Currently not used
 # since we need configuration.
 # --iteration 1 adds a "-1" suffix to the version that didn't exist before
-${ISTIO_OUT}/release/istio-sidecar.deb: | ${ISTIO_OUT} deb/fpm
+${ISTIO_OUT_LINUX}/release/istio-sidecar.deb: | ${ISTIO_OUT_LINUX} deb/fpm
 
 # Package the sidecar deb file.
 deb/fpm:
-	rm -f ${ISTIO_OUT}/release/istio-sidecar.deb
-	fpm -s dir -t deb -n ${ISTIO_DEB_NAME} -p ${ISTIO_OUT}/release/istio-sidecar.deb --version $(PACKAGE_VERSION) -f \
+	rm -f ${ISTIO_OUT_LINUX}/release/istio-sidecar.deb
+	fpm -s dir -t deb -n ${ISTIO_DEB_NAME} -p ${ISTIO_OUT_LINUX}/release/istio-sidecar.deb --version $(PACKAGE_VERSION) -f \
 		--url http://istio.io  \
 		--license Apache \
 		--vendor istio.io \
@@ -71,9 +71,9 @@ deb/fpm:
 		--depends iptables \
 		$(SIDECAR_FILES)
 
-${ISTIO_OUT}/istio.deb:
-	rm -f ${ISTIO_OUT}/release/istio.deb
-	fpm -s dir -t deb -n istio -p ${ISTIO_OUT}/release/istio.deb --version $(PACKAGE_VERSION) -f \
+${ISTIO_OUT_LINUX}/istio.deb:
+	rm -f ${ISTIO_OUT_LINUX}/release/istio.deb
+	fpm -s dir -t deb -n istio -p ${ISTIO_OUT_LINUX}/release/istio.deb --version $(PACKAGE_VERSION) -f \
 		--url http://istio.io  \
 		--license Apache \
 		--vendor istio.io \
@@ -82,14 +82,14 @@ ${ISTIO_OUT}/istio.deb:
 		$(ISTIO_FILES)
 
 # Install the deb in a docker image, for testing of the install process.
-deb/docker: build deb/fpm ${ISTIO_OUT}/release/istio.deb
+deb/docker: build deb/fpm ${ISTIO_OUT_LINUX}/release/istio.deb
 	mkdir -p ${OUT_DIR}/deb
 	cp tools/packaging/deb/Dockerfile tools/packaging/deb/deb_test.sh ${OUT_DIR}/deb
 	cp tests/testdata/config/*.yaml ${OUT_DIR}/deb
 	cp -a tests/testdata/certs ${OUT_DIR}/deb
 	cp ${GOPATH}/bin/{kube-apiserver,etcd,kubectl} ${OUT_DIR}/deb
-	cp ${ISTIO_OUT}/istio-sidecar.deb ${OUT_DIR}/deb/istio-sidecar.deb
-	cp ${ISTIO_OUT}/istio.deb ${OUT_DIR}/deb/istio.deb
+	cp ${ISTIO_OUT_LINUX}/istio-sidecar.deb ${OUT_DIR}/deb/istio-sidecar.deb
+	cp ${ISTIO_OUT_LINUX}/istio.deb ${OUT_DIR}/deb/istio.deb
 	docker build -t istio_deb -f ${OUT_DIR}/deb/Dockerfile ${OUT_DIR}/deb/
 
 deb/test:

